src/lib/operators/join_sort_merge.cpp

```diff
@@ -81,12 +84,14 @@ template <typename T>
 class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
  public:
   JoinSortMergeImpl<T>(JoinSortMerge& sort_merge_join, ColumnID left_column_id, ColumnID right_column_id,
-                       const PredicateCondition op, JoinMode mode)
+                       const PredicateCondition op, JoinMode mode,
+                       const std::vector<OperatorJoinPredicate>& additional_join_predicates)
```

Comment:
It's okay that you add this member (pls rename it to "secondary_join_predicates", though), but pls Assert that it is empty so no one creates broken SortMergeJoins until the SortMergeJoin supports MPJ.

Or is adding MPJ support to SMJ planned to be done as part of this PR?